Scheduler test can run into a data race error.

Details:

```
2018-08-15T21:47:24.501+0200 [DEBUG] Gaia: error during job execution: error="job failed" job="{2296842551 Job1  [] running}"
panic: send on closed channel

goroutine 13 [running]:
github.com/gaia-pipeline/gaia/scheduler.(*Scheduler).resolveDependencies(0xc420020f00, 0x89e70aca, 0x11e6932, 0x4, 0x0, 0x0, 0xc42000e2d0, 0x1, 0x1, 0x11ea8ba, ...)
        .../golang/src/github.com/gaia-pipeline/gaia/scheduler/scheduler.go:376 +0x2f0
github.com/gaia-pipeline/gaia/scheduler.(*Scheduler).resolveDependencies(0xc420020f00, 0x8ae70c5d, 0x11e6936, 0x4, 0x0, 0x0, 0xc42000e2d8, 0x1, 0x1, 0x11ea8ba, ...)
        .../golang/src/github.com/gaia-pipeline/gaia/scheduler/scheduler.go:358 +0x223
created by github.com/gaia-pipeline/gaia/scheduler.(*Scheduler).executeScheduler
        ...golang/src/github.com/gaia-pipeline/gaia/scheduler/scheduler.go:429 +0x2c9
```

Circle CI link: https://circleci.com/gh/gaia-pipeline/gaia/457?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

--- a/scheduler/scheduler.go
+++ b/scheduler/scheduler.go
@@ -522,6 +522,17 @@ func (s *Scheduler) executeScheduler(r *gaia.PipelineRun, pS Plugin) {
 				// Send done signal to all resolvers
 				close(done)
 
+				// read all jobs which are waiting to be executed to free the channel
+				var channelClean = false
+				for !channelClean {
+					select {
+					case <-executeScheduler:
+						// just read from the channel
+					default:
+						channelClean = true
+					}
+				}
+
 				// Close executeScheduler. No new jobs should be scheduled.
 				close(executeScheduler)
 

